src/views/v2/commerce/checkout/order-modifiers/coupons.php

```diff
@@ -3,32 +3,77 @@
  * Logic for handling coupons during the checkout process.
  *
  * @since   5.18.0
- * @package TEC\Tickets\Commerce\Order_Modifiers\Checkout
+ * @since   TBD Updated with actual coupon logic.
  *
- * @private Still a work in progress.
+ * @package TEC\Tickets\Commerce\Order_Modifiers\Checkout
  *
- * @phpcs:disable
- * @todo: Add the necessary logic to handle coupons during the checkout process.
+ * @var array[] $items [Global] List of Items on the cart to be checked out.
  */
 
+use TEC\Tickets\Commerce\Utils\Value;
+
+$coupon_input_classes = [
+	'tribe-tickets__commerce-checkout-cart-coupons__input',
+	'tribe-common-form-control-text__input',
+	'tribe-tickets__form-field-input',
+];
+
+$apply_button_classes = [
+	'tec-tickets__commerce-checkout-cart-coupons__apply-button',
+	'tribe-common-c-btn',
+];
+
+$coupon_items = array_filter( $items, fn( $item ) => 'coupon' === ( $item['type'] ?? '' ) );
+
+// If we have any coupons, always use the first one. There *shouldn't* be more than one.
+$coupon = ! empty( $coupon_items ) ? $coupon_items[ array_key_first( $coupon_items ) ] : [];
+
+// Determine the discount to display.
+$discount = '';
+if ( isset( $coupon['sub_total'] ) && $coupon['sub_total'] instanceof Value ) {
+	$discount = $coupon['sub_total']->get_currency();
+}
+
+// Sections that should be hidden based on the coupon state.
+$hide_input   = ! empty( $coupon );
+$hide_applied = empty( $coupon );
```

Comment:
Why two values that are opposites - just use one and compare `$hide_input` or `! $hide_input`

Or, (actually, _also_) add a class instead of an inline-style. Use `tribe_classes()` with conditionals.

i.e.
```
tribe_classes(
    [
        'tec-tickets__commerce-checkout-cart-coupons' => true
        'hide-that-sucker' => empty( $coupon )
    ]
);